### ISSUE TITLE
Woody crops in CLM

### ISSUE THREAD

**Issue by ashehad** [ESCOMP/CTSM#1908]
Dear All, 

I would like to give an update about the crop developments that Yuanchao Fan and I have been doing in CLM. We have been focusing on perennial woody crop PFTs in CLM5 (oil palm and rubber), but with a view to being applicable to several other crops of similar type. There are numerous woody crop plantations around the globe, some as monocultures such as cocoa, coffee, and coconut palms while others as agroforestry plantations such as jungle rubber plantations (rubber trees with native trees) and coffee plantations with native trees. I would also like to update the key features we have developed in the existing crop phenology scheme (e.g., perennial, evergreen/deciduous, woody, monthly harvests, annual harvest). We have been focusing on developing/extending the default crop phenology subroutine and keeping it as generic as possible.

Currently, I am calibrating these two crop PFTs. We plan to submit a merge request to NCAR's CTSM trunk soon.

Best,
Ashehad

**Comment by wwieder** [comment]
This also seems to be related to work on CLM-Fruit Tree that Olga's doing for apple orchard simulations.  I'm not sure what long-term plans or resources there are to support woody , perennial trees in the crop model?

**Comment by ashehad** [comment]
Yes, this might be related to Olga's work. I do not have long-term plans yet, but at the moment I can get ctsm running in docker. At present, we have some ground measurements ongoing for perennial tree crops in the tropics.

**Comment by odombro** [comment]
Hi, yes thanks for mentioning. I just openend a new issue on the fruit tree development #1956

**Comment by ashehad** [comment]
Dear All,

Please find below a recent version of clm5 (ctsm1.0.dev040) that has relevant codes for oil palm and rubber that I mentioned in the first thread:

https://github.com/yfa008/clm-palm-dev/tree/clm-palm-dev01-ctsm1.0.dev040

Best,
Ashehad

**Comment by wwieder** [comment]
Thanks for pointing us to this work, @ashehad and @yfa008.  It's exciting to see these capabilities being developed by the community!

We need to have a broader conversation about if and how we can support the development of perennial/woody crops in CTSM given limited software engineering resources.

One thing that would be helpful would be to bring merge your development work up to a current CTSM5.1 dev tag and test that it's still working as intended.  Is this something that you are able to do, Ashehad or Yuanchao?

**Comment by ashehad** [comment]
Yes, we'd definitely like to merge it to the current tag i.e CTSM5.1 dev tag.
